File: 9-palindrome-number/palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x<0:
            return False
        
        if x<10:
            return True
            
        st = str(x)
        s = ''
        while x>0:
            r = x%10
            s += str(r)
            x//=10

        if st==s:
            return True
        else:
            return False 


        

File: 9-palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x<0){
            return false;
        }
        int temp = x;
        int res = 0;
        while(temp>0){
            int rem = temp%10;
            res = res*10 + rem;
            temp/=10;
        }
        return res==x;
    }
}